### PR TITLE
Use build args for JAVA_OPTS and CATALINA_OPTS (#587)

### LINKED ISSUE #587: Proxy problem: Couldn't process file data/osm_file.pbf, error: java.net.SocketTimeoutException
Hi,
I'm having trouble launching the app via the docker-compose config.

I've been able to build the image after git cloning and docker-compose brings the server up, but it never enters the ready state. Instead, shortly after launching, I get a long list of Java stack traces, summarized in the ors.log file as:

2019-09-10 20:08:37,681 ERROR [routing.RoutingProfileManager] - java.util.concurrent.ExecutionException: java.lang.RuntimeException: Couldn't process file data/osm_file.pbf, error: java.net.SocketTimeoutException: connect timed out
2019-09-10 20:08:40,079 ERROR [routing.RoutingProfileManager] - java.util.concurrent.ExecutionException: java.lang.RuntimeException: Couldn't process file data/osm_file.pbf, error: Index: 7754600, Size: 0
2019-09-10 20:08:42,377 ERROR [routing.RoutingProfileManager] - java.util.concurrent.ExecutionException: java.lang.RuntimeException: Couldn't process file data/osm_file.pbf, error: Index: 7754600, Size: 0
2019-09-10 20:08:44,573 ERROR [routing.RoutingProfileManager] - java.util.concurrent.ExecutionException: java.lang.RuntimeException: Couldn't process file data/osm_file.pbf, error: Index: 7754600, Size: 0

I have verified that osm_file.pbf is present in the Docker container. I'm not sure what a socket timeout on reading a file means.

The data and configuration files I am using are unmodified from the initial git clone. The only difference I have made is modifying MAVEN_OPTS in the Dockerfile to add the corporate proxy that I am working behind.

I'd appreciate any advice on the subject.

### PR DESCRIPTION
**Pull request by stevecambridge**
### Pull Request Checklist

- [x] 1. I have [**rebased**](https://github.com/GIScience/openrouteservice/blob/master/CONTRIBUTE.md#pull-request-guidelines) the latest version of the development branch into my feature branch and all conflicts have been resolved.
- [x] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the [Unreleased] heading.
- [x] 3. I have documented my code using JDocs tags.
- [x] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [x] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [x] 6. I have created API tests for any new functionality exposed to the API.
- [x] 7. If changes/additions are made to the app.config file, I have added these to the app.config wiki page on github along with a short description of what it is for, and documented this in the Pull Request (below).
- [ ] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing *(no API changes)*
- [x] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [x] 10. For new features or changes involving building of graphs, I have tested on a larger dataset (at least Germany) and the graphs build without problems (i.e. no out-of-memory errors).
- [x] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the importer etc.), I have generated longer distance routes for the affected profiles with different options (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end points generated from the current live ORS. If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [x] 12. I have written in the Pull Request information about the changes made including their intended usage and why the change was needed.

Fixes #587  .

### Information about the changes

Moves JAVA_OPTS and CATALINA_OPTS from the "environment" section of docker-compose.yml to the build arguments section. They're declared as arguments in the Dockerfile anyway, and the RUN echo lines copy them into an environment setup script on build. If these values aren't passed as build args, the setup script unsets those variables by running JAVA_OPTS="" and CATALINA_OPTS="" when the container is run.

### Examples and reasons for differences between live ORS routes and those generated from this pull request
-

### Required changes to app.config (if applicable)
-

### CODE REVIEW

**Review by rabidllama — APPROVED** [review]

**Comment by rabidllama** [comment]
Hi @stevecambridge 

something is going a bit weird atm in that the build checks are failing since I updated the branch to the latest development. we did a merge yesterday of a sonarqube integration step and something is causing an issue in this PR, but not in others. Would it be possible for you to clear the merge commit that I did (a thing github does), rebase from the latest development and then force push your changes to your repo? Sorry for the pain, but I have no idea why it is not playing nicely

Adam

**Comment by stevecambridge** [comment]
@rabidllama No worries, I've reset it and force pushed.

**Comment by stevecambridge** [comment]
Still seems unhappy. Not sure what to make of the whole incredibly long Travis output but seems like there's an issue reaching your sonarqube server:
```
[ERROR] SonarQube server [http://localhost:9000] can not be reached

[INFO] ------------------------------------------------------------------------

[INFO] BUILD FAILURE

[INFO] ------------------------------------------------------------------------

[INFO] Total time: 01:06 min

[INFO] Finished at: 2019-11-12T14:28:10Z

[INFO] Final Memory: 68M/553M

[INFO] ------------------------------------------------------------------------

[ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.7.0.1746:sonar (default-cli) on project openrouteservice: Unable to execute SonarQube: Fail to get bootstrap index from server: Failed to connect to localhost/127.0.0.1:9000: Connection refused (Connection refused) -> [Help 1]
```

**Comment by rabidllama** [comment]
I am reverting a merge we did the other day for the sonarqube thing - it seems it is causing problems with some PR as another one is failing with the same thing... Will try and get it updated when that is done

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -9,12 +9,11 @@ services:
             args:
                 APP_CONFIG: ./docker/conf/app.config.sample
                 OSM_FILE: ./docker/data/heidelberg.osm.gz
+                JAVA_OPTS: -Dhttp.proxyHost=10.64.1.100 -Dhttp.proxyPort=80 -Dhttps.proxyHost=10.64.1.100 -Dhttps.proxyPort=443 -Djava.awt.headless=true -server -XX:TargetSurvivorRatio=75 -XX:SurvivorRatio=64 -XX:MaxTenuringThreshold=3 -XX:+UseConcMarkSweepGC -XX:+UseParNewGC -XX:ParallelGCThreads=4 -Xms1g -Xmx2g
```
Comment (rabidllama) [inline docker/docker-compose.yml]: are the proxy host and port attributes required?

Comment (stevecambridge) [reply docker/docker-compose.yml]: Whoops - no, I left that in by mistake. Sorry about that.

**Review by rabidllama — APPROVED** [review]

**Pull request merged**